tests/unit_tests/test_league_rater.py

```diff
@@ -0,0 +1,363 @@
+        GameOutcome.VICTORY,
+        player_rating,
+    )
+
+    assert new_score.division_id == current_score.division_id
+    assert new_score.game_count == current_score.game_count + 1
+    assert new_score.score == current_score.score + config.SCORE_GAIN + config.HIGHEST_DIVISION_BOOST
+
+
+def test_placement_after_enough_games(example_league, unplaced_player_score):
+    # Neutralize the offset in the placement function so we can test the score independently of the config settings
+    rating = (150.0 - config.RATING_MODIFIER_FOR_PLACEMENT, 0.0)
+
+    new_score = LeagueRater.rate(example_league, unplaced_player_score, None, rating)
+
+    assert new_score.division_id == example_league.divisions[1].id
+    assert new_score.game_count == unplaced_player_score.game_count + 1
+    assert new_score.score == 5
+
+
+def test_replacement_at_invalid_player_division(example_league):
+    current_score = LeagueScore(division_id=999, score=4, game_count=config.PLACEMENT_GAMES)
+    rating = (150.0 - config.RATING_MODIFIER_FOR_PLACEMENT, 0.0)
+
+    new_score = LeagueRater.rate(example_league, current_score, None, rating)
+
+    assert new_score.division_id == example_league.divisions[1].id
+    assert new_score.game_count == current_score.game_count + 1
+    assert new_score.score == 5
+
+
+def test_placement(example_league, unplaced_player_score):
+    # Neutralize the offset in the placement function so we can test the score independently of the config settings
+    rating = 150 - config.RATING_MODIFIER_FOR_PLACEMENT
+
+    new_score = LeagueRater._do_placement(example_league, unplaced_player_score, rating)
+
+    assert new_score.division_id == example_league.divisions[1].id
+    assert new_score.game_count == unplaced_player_score.game_count + 1
+    assert new_score.score == 5
+
+
+def test_placement_high_rating(example_league, unplaced_player_score):
+    rating = 1500 - config.RATING_MODIFIER_FOR_PLACEMENT
+
+    new_score = LeagueRater._do_placement(example_league, unplaced_player_score, rating)
+
+    assert new_score.division_id == example_league.divisions[-1].id
+    assert new_score.game_count == unplaced_player_score.game_count + 1
+    assert new_score.score == 10
+
+
+def test_placement_low_rating(example_league, unplaced_player_score):
+    rating = -500 - config.RATING_MODIFIER_FOR_PLACEMENT
+
+    new_score = LeagueRater._do_placement(example_league, unplaced_player_score, rating)
+
+    assert new_score.division_id == example_league.divisions[0].id
+    assert new_score.game_count == unplaced_player_score.game_count + 1
+    assert new_score.score == 0
+
+
+def test_new_player(example_league):
+    current_score = LeagueScore(division_id=None, score=None, game_count=0)
+    player_rating = (380.0, 0.0)
+
+    new_score = LeagueRater.rate(
+        example_league,
+        current_score,
+        GameOutcome.VICTORY,
+        player_rating,
+    )
+
+    assert new_score.division_id == current_score.division_id
```

Comment:
I guess here I would actually use `None`, since "player hasn't been placed" is more comprehensible than "player wasn't placed before and the division hasn't changed" :stuck_out_tongue_closed_eyes: 
```suggestion
    assert new_score.division_id is None
```